components/SmartContract.tsx

```diff
@@ -2,79 +2,161 @@ import { useTranslation } from 'next-i18next'
 import NextLink from 'next/link'
 import Tooltip from 'components/Tooltip'
 import InfoList from './InfoList'
-import { AddModeratorOutlined as RegisterIcon } from '@mui/icons-material'
-import { GraphQLSchema } from 'utils'
+import { client, GraphQLSchema } from 'utils'
+import NextPageIcon from 'assets/icons/next-page.svg'
+import VerifiedIcon from 'assets/icons/check-success.svg'
+import SubmittedIcon from 'assets/icons/submit-success.svg'
+import { Skeleton } from '@mui/material'
+import { useState } from 'react'
+import { gql } from 'graphql-request'
 
 const CONTRACT_FORM_URL = `https://github.com/Magickbase/godwoken_explorer/issues/new/choose`
 
+const verifyAndUpdateFromSourcify = gql`
+  mutation verifyAndUpdateFromSourcify($address: HashAddress!) {
+    verify_and_update_from_sourcify(input: { address: $address }) {
+      account_id
+    }
+  }
+`
+
+export const recheckVerifyStatus = (address: string) =>
+  client
+    .request<{ verify_and_update_from_sourcify: { account_id: string | null } }>(verifyAndUpdateFromSourcify, {
+      address: address,
+    })
+    .then(data => data.verify_and_update_from_sourcify.account_id)
+
 const SmartContract: React.FC<{
   deployer: string
   deployTxHash: string
   udt: Pick<GraphQLSchema.Udt, 'id' | 'name'> | null
   isVerified: boolean
-}> = ({ deployer, deployTxHash, udt, isVerified }) => {
+  isSubmitted: boolean
+  refetchStatus: any
+  address: string
+}> = ({ deployer, deployTxHash, udt, isVerified, isSubmitted, refetchStatus, address }) => {
   const [t] = useTranslation('account')
+  const [checkAgain, setCheckAgain] = useState(false)
+  const [isSourcifyCheckLoading, setIsSourcifyCheckLoading] = useState(false)
+
+  const submmitedInfo = isSubmitted
```

Comment:
deployer address and deploy tx hash are determined on chain so condition `isSubmitted` is unnecessary.

Reply:
OK, how to check when to display this icon?
<img width="540" alt="image" src="https://user-images.githubusercontent.com/32790369/186057388-8be21d67-555d-41e1-8155-5d5ef78436cf.png">

Reply:
> OK, how to check when to display this icon? <img alt="image" width="540" src="https://user-images.githubusercontent.com/32790369/186057388-8be21d67-555d-41e1-8155-5d5ef78436cf.png">

As token https://github.com/Magickbase/godwoken_explorer/issues/877 shows, if it has one of `UAN`, website, logo, description(they are all off-chain data and can only be updated by registration), its information has been submitted.